src/ngx_http_lua_headers.c

```diff
@@ -480,12 +483,25 @@ ngx_http_lua_ngx_req_get_headers(lua_State *L)
                        "lua request header: \"%V: %V\"",
                        &header[i].key, &header[i].value);
 
-        if (--count == 0) {
+        if (--count <= 0) {
```

Comment:
Will this be clearer:

```C
for (...) {
    if (--count <= 0) {
        break;
    }
}

if (truncated) {
    goto truncated;
}

return 1;

truncated:

lua_pushliteral(L, "truncated");
return 2;
```

Similar code can be found in `ngx_http_lua_ngx_resp_get_headers` as well.